core/modules/layout/includes/custom_text_block.inc

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ *  CustomTextBlock extends Block
+ *  
+ *  This class allows us to create custom text blocks.
+ */
+class CustomTextBlock extends Block {
+  /**
+   *  Sets title text on draggable block panel in Layout builder.
+   */
+  function getAdminTitle() {
+    return t('Custom text block');
+  }
+
+  /**
+   *  Sets description text on draggable block in Layout builder.
+   */
+  function getAdminPreview() {
+    $label_text = !empty($this->settings['subject']) ? ': ' . check_plain($this->settings['subject']) : '';
+    $preview = '<p>Custom text' . $label_text;
+    $preview .= '</p>';
+    
+    if (!empty($this->conditions)) {
+      foreach ($this->conditions as $condition) {
+        $condition_list[] = $condition->summary();
+      }
+      $conditions = theme('item_list', array('items' => $condition_list, 'attributes' => array('class' => array('block-conditions'))));
+      $fieldset = array(
+        '#type' => 'fieldset',
+        '#collapsible' => TRUE,
+        '#collapsed' => TRUE,
+        '#title' => t('Conditions'),
+        '#children' => $conditions,
+        '#parents' => array(),
+      );
+      $form_state = array();
+      form_process_fieldset($fieldset, $form_state);
+      $preview .= backdrop_render($fieldset);
+    }
```

Comment:
This looks good, but it's a lot of copy/paste from the Block::getAdminPreview() function. Perhaps we should split it up so that child-blocks can inherit or call a parent function to reduce duplication.

What if we had something like this:

```
function getAdminPreview() {
   $label_text = !empty($this->settings['subject']) ? ': ' . check_plain($this->settings['subject']) : '';
   $preview = '<p>Custom text' . $label_text;
   $preview .= '</p>';
   $preview .= $this->getAdminConditionsPreview();
```

Where getAdminConditionsPreview() was a new protected function in the parent `Block` class?